### PR TITLE
Fix inability to drag files inside folders #3550

### PR DESCRIPTION
**Pull request by pnobre**

### CODE REVIEW

**Comment by cartermp** [comment]
Linking #3550 via GitHub and renaming the PR to say it fixes the issue

File: vsintegration/src/FSharp.ProjectSystem.Base/Project/ProjectNode.CopyPaste.cs [hunk 1]
```diff
@@ -39,11 +39,6 @@ public override int DragEnter(IOleDataObject pDataObject, uint grfKeyState, uint
         {            
             pdwEffect = (uint)DropEffect.None;
 
-            if (this.SourceDraggedOrCutOrCopied)
```
Comment (cartermp) [inline vsintegration/src/FSharp.ProjectSystem.Base/Project/ProjectNode.CopyPaste.cs]: @KevinRansom you're shown as the last person to touch this part of the code...any concerns?

**Comment by KevinRansom** [comment]
Well ...

last time we touched it, the project file order got out of sync with the UI due to copy and paste.  So we will need to do a bunch of interesting manual testing.

Kevin

**Comment by cartermp** [comment]
Going through @saul's checklist here: https://github.com/Microsoft/visualfsharp/pull/2692

* Move up/down
    * Folder moves all of its contents ✔️ 
    * Subfolder moves all of its contents ✔️ 
    * Project file and solution explorer are correct ✔️ 
* Folder node selected
    * Add above --> new/existing/folder ✔️ 
    * Add below --> new/folder ✔️ 
    * Add above/below on an empty folder skips the folder (not adding into it) ✔️ 
    * Add existing file **that is in not in the folder's directory** ❌ 
* File node selected
    * Add above/below --> new within subfolder ✔️ 
    * Add above/below --> existing file that is within the current subfolder ✔️ 
    * Add above/below --> existing file that is **not** within the current subfolder ❌
* Cut/Copy and Paste
    * Copy from subfolder to subfolder ✔️ 
    * Cut from subfolder to subfolder ✔️ 
    * Solution explorer and fsproj ordering are matched ✔️ 
* Add existing
    * On folder node should place the file in the correct subfolder ✔️ 
    * On project node should add folders/subfolders as needed ✔️ 
* Add Folder ✔️ 

The two ❌s are likely the same root problem, but they do manifest themselves differently:

1. New .NET Framework console app
2. Add new folder
3. Add new file in that folder
4. Outside of visual studio, create a new file in the project root directory.
5. Select the folder node and add the newly-created file as an existing file.

**Expected:** It is added within the subfolder.
**Actual:** It is added to the project root.

6. Add a new file within the subfolder.
7. Outside of visual studio, create a new file in the project root directory.
8. Select the new file in the subfolder, select "Add Below" > "Existing Item"
9. Select the file you created in the project root directory.

**Expected:** The file is added to the subfolder.
**Actual:** Modal dialog saying there is an error due to the file not being in the same directory, then it is added to the project root.

dang
dang-root

In C#, adding an existing file that is not in the subdirectory you wish to add to will add it there, so I would say there is a bug here.  However, I do not think that PR is the cause.  @saul, what do you think?

**Pull request merged**